spec/lib/puppet_spec/files.rb

```diff
@@ -43,6 +43,20 @@ def self.file_containing(name, contents)
     file
   end
 
+  def script_containing(name, contents) PuppetSpec::Files.script_containing(name, contents) end
+  def self.script_containing(name, contents)
+    file = tmpfile(name)
+    if Puppet.features.microsoft_windows?
+      file += '.bat'
+      text = contents[:windows]
+    else
+      text = contents[:posix]
+    end
+    File.open(file, 'w') { |f| f.write(text) }
```

Comment:
Since ruby defaults to text mode on windows, it will translate "\n" to "\r\n", which I think is ok for scripts. But if this helper is ever used (in the future) to create a file whose checksum we care about, then we might be surprised. Maybe default to 'wb' always, or allow the caller to specify the text/binary mode?